File: Sprint1/C++/C/main.cpp
// https://contest.yandex.ru/contest/22449/problems/C/
#include <iostream>
#include <vector>
#include "algorithm"

int main() {
    int n, m;

    std::cin >> n;
    std::cin >> m;

    std::vector<std::vector<int>> matrix(n, std::vector<int>(m));
    std::vector<int> neighbors;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            std::cin >> matrix[i][j];
        }
    }

    int row, col;
    std::cin >> row >> col;

    if (col - 1 >= 0) {
        neighbors.push_back(matrix[row][col - 1]);
    }
    if (col + 1 < m) {
        neighbors.push_back(matrix[row][col + 1]);
    }
    if (row - 1 >= 0) {
        neighbors.push_back(matrix[row - 1][col]);
    }
    if (row + 1 < n) {
        neighbors.push_back(matrix[row + 1][col]);
    }

    std::sort(neighbors.begin(), neighbors.end());

    for (int neighbor: neighbors) {
        std::cout << neighbor << " ";
    }
    std::cout << std::endl;

    return 0;
}

File: Sprint1/C/Rust/src/main.rs
// https://contest.yandex.ru/contest/22449/problems/C/

use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let m: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();

    let mut matrix = vec![vec![0; m]; n];

    for matrix_row in matrix.iter_mut() {
        let row: Vec<i32> = lines
            .next()
            .unwrap()
            .unwrap()
            .split_whitespace()
            .map(|s| s.parse().unwrap())
            .collect();
        *matrix_row = row;
    }

    let row: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let col: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();

    let mut neighbors = vec![];

    if col > 0 {
        neighbors.push(matrix[row][col - 1]);
    }
    if col + 1 < m {
        neighbors.push(matrix[row][col + 1]);
    }
    if row > 0 {
        neighbors.push(matrix[row - 1][col]);
    }
    if row + 1 < n {
        neighbors.push(matrix[row + 1][col]);
    }

    neighbors.sort();
    let output = neighbors
        .iter()
        .map(|&x| x.to_string())
        .collect::<Vec<String>>()
        .join(" ");

    println!("{}", output);
}
